[CI] GoogleCloudStorageBlobContainerRetriesTests testRetriesAreTerminatedWhenClientProviderIsClosed failing

**Build Scans:**
- [elasticsearch-periodic #11869 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/gz4hg6ng4njbw)
- [elasticsearch-periodic #11869 / openjdk21_checkpart1_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/v6yzui7ntedse)
- [elasticsearch-periodic #11869 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/zb3rqsovpephc)

**Reproduction Line:**
```
./gradlew ":modules:repository-gcs:test" --tests "org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests.testRetriesAreTerminatedWhenClientProviderIsClosed" -Dtests.seed=2BE914A890913301 -Dtests.locale=en-DE -Dtests.timezone=Europe/Ljubljana -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRetriesAreTerminatedWhenClientProviderIsClosed'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected org.apache.lucene.store.AlreadyClosedException to be thrown, but nothing was thrown
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testRetriesAreTerminatedWhenClientProviderIsClosed
- [main] 3 failures in test testRetriesAreTerminatedWhenClientProviderIsClosed (6.0% fail rate in 50 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java
+++ b/modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java
@@ -653,7 +653,10 @@ public class AzureBlobContainerRetriesTests extends AbstractBlobContainerRetries
             }
         });
 
-        assertThrows(AlreadyClosedException.class, () -> blobContainer.readBlob(randomRetryingPurpose(), "read_blob_while_store_closes"));
+        assertThrows(
+            AlreadyClosedException.class,
+            () -> Streams.readFully(blobContainer.readBlob(randomRetryingPurpose(), "read_blob_while_store_closes"))
+        );
     }
 
     private BlobContainer createBlobContainer(int maxRetries, String secondaryHost, LocationMode locationMode) {

--- a/modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java
+++ b/modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java
@@ -706,7 +706,10 @@ public class GoogleCloudStorageBlobContainerRetriesTests extends AbstractBlobCon
             }
         });
 
-        assertThrows(AlreadyClosedException.class, () -> blobContainer.readBlob(randomRetryingPurpose(), "read_blob_while_store_closes"));
+        assertThrows(
+            AlreadyClosedException.class,
+            () -> Streams.readFully(blobContainer.readBlob(randomRetryingPurpose(), "read_blob_while_store_closes"))
+        );
     }
 
     private HttpHandler safeHandler(HttpHandler handler) {

